### ISSUE TITLE
Schedule and do DevOps maintenance tasks

### ISSUE THREAD

**Issue by mddilley** [cityofaustin/atd-data-tech#19797]
We have discussed making a ritual out of keeping various services and software to mimic the continuous updates we make to Hasura during VZE and Moped releases. This would involve creating a launch checklist with each item below that we would use as a template for the next scheduled batch of updates. We should assign to a couple of us to pair and get it off the ground. Then, we can create a whole-team release party

**Tasks:**
- [x] Upgrade Airflow (3 estimate) ([260](https://github.com/cityofaustin/atd-airflow/pull/260) and [261](https://github.com/cityofaustin/atd-airflow/pull/260))
   - Currently on v2.9.0
    - **Steps**: Read [release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html), locally test build, PR the change, merge to production after approvals, read diffs of config changes and preserve previous config (look to documentation of software that has config diff)

- [x] All EC2 host system upgrade (0 estimate)
   - Goal: Get to current version
   - **Steps**: High severity updates will be applied, `sudo aptitude` to review and accept changes, Docker service will restart (breaks connections)
   1. Get on the machine
   2. sudo apt update
   3. sudo apt list --upgradable; # Use this to look at the list of packages that are going to be updated; look at them to see if anything sticks out. Look for docker in particular; are you ok if the service restarts for a few seconds?
   4. sudo apt upgrade
   5. check that everything is working
   - [x] Bastion host: Upgraded ✅ (test connection to read replicas)
   - [x] Citybase - Note: ~1GB of disk space left
   - [x] EAZ; waiting on super user access
   - [x] postgrest: Just created ✅
   - **Resource**: https://packages.ubuntu.com/
- [x] Upgrade postgREST
   - **Steps**: 
      1. Backup postgrest
      2. SSH into postgrest EC2
      3. `sudo -s`
      4. cd /srv/atd-postgrest
      5. docker compose pull
      6. docker compose restart (or down and up, or however you want to restart it)
   - ~Incorporate after completing https://github.com/cityofaustin/atd-data-tech/issues/19268~
   - What about postgREST RDS instance upgrades?
- [x] Create a run of show
   - https://github.com/cityofaustin/atd-data-tech/issues/20105  
- [x] Document each task so that anyone on the team could accomplish this
   - https://app.gitbook.com/o/-LzDQOVGhTudbKRDGpUA/s/-M4Ul-hSBiM-3KkOynqS/devops-maintenance-tasks 
- [x] Evaluate the total level of effort and how that fits into targeting a monthly schedule

**Out of Scope**
- 1PW server updates are captured in https://github.com/cityofaustin/atd-data-tech/issues/15249

**Comment by mddilley** [comment]
I created the [first run of show here](https://github.com/cityofaustin/atd-data-tech/issues/20105), and there are [docs here](https://app.gitbook.com/o/-LzDQOVGhTudbKRDGpUA/s/-M4Ul-hSBiM-3KkOynqS/devops-maintenance-tasks).

I'll ask for some feedback on the docs early next week after the holiday.

**Comment by frankhereford** [comment]
👋 Hi @mddilley - we're getting close on putting a pin in this issue. 😅 I got the EC2 instances up to date today, and updated the main issue. Thanks!

**Comment by mddilley** [comment]
Frank and I ran through the Airflow stack update yesterday, and everything came back up so the last checkbox is covered. I created one last PR that isn't urgent but contains some small changes that came from me go step by step through the current readme. https://github.com/cityofaustin/atd-airflow/pull/263

We are now ready for the next go-around! 🙌 🚀 😎
